### ISSUE TITLE
Repo: Sponsor image 404s cause website tests to fail

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#7332]
### Suggestion

Irksome. Example in https://github.com/typescript-eslint/typescript-eslint/actions/runs/5670711873/job/15366549313:

```plaintext
  1) [chromium] › index.spec.ts:10:7 › Website › should have no errors ─────────────────────────────

    Error: expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 3

    - Array []
    + Array [
    +   "[error] Failed to load resource: the server responded with a status of 404 ()",
    + ]
```

The issue is that a sponsor's image is a 404 and our `sponsors.json` file hasn't been updated yet.

Fixed by @rubiesonthesky in #7247's `packages/website/tests/index.spec.ts` changes: https://github.com/typescript-eslint/typescript-eslint/pull/7247/files#diff-227115ad1ec8da7528470f6ee7b0b0d9bd3adec565656c2fffedc27274cf7ac8

```js
test.beforeEach(async ({ context }) => {
  // Sponsor logos are sometimes changed or removed between deploys
  await context.route('https://images.opencollective.com/**/*.png', route =>
    route.fulfill({
      status: 200,
      body: '',
    }),
  );
});
```
